Composer/packages/tools/language-servers/language-generation/src/LGServer.ts

```diff
@@ -239,13 +268,15 @@ export class LGServer {
       const content = this.documents.get(uri)?.getText() || '';
       // if inline mode, composite local with server resolved file.
       const lgTextFiles = projectId ? this.getLgResources(projectId) : [];
+      this.recordTemplatesDefintions(lgTextFiles);
```

Comment:
I dont think this is the best place to call this, just because you have access to the right arguments.

Reply:
Righit, it should called at the same stage as addLGDocument, I will change that.

Reply:
moved to initializedocument body